framework/include/distributions/Distribution.h

```diff
@@ -0,0 +1,65 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifndef DISTRIBUTION_H
+#define DISTRIBUTION_H
+
+#include "MooseObject.h"
+#include "MooseRandom.h"
+
+class Distribution;
+
+template<>
+InputParameters validParams<Distribution>();
+
+/**
+ * All Distributions should inherit from this class
+ */
+class Distribution :
+  public MooseObject
+{
+public:
+  Distribution(const InputParameters & parameters);
+  virtual ~Distribution();
+  /**
+   * Compute the pdf value for given variable value x
```

Comment:
Can you give better descriptions in your methods? I have no idea what "pdf" and "cdf" stand for.